### PR TITLE
Log outstanding writes in DynamoDB

### PR DESCRIPTION
**Pull request by bboreham**
To help investigate #733.  From existing log lines I know we very commonly get just 1 item remaining at the timeout, so I want to know what it is.

Also some drive-by comments because I needed to figure out what was going on.

### CODE REVIEW

File: pkg/chunk/aws/storage_client.go [hunk 1]
```diff
@@ -731,6 +734,7 @@ func (a storageClient) putS3Chunk(ctx context.Context, key string, buf []byte) e
 	})
 }
 
+// Slice of values returned; map key is attribute name
```
Comment (jml) [inline pkg/chunk/aws/storage_client.go]: Thanks for this & similar comments. What do you think about a follow-up PR replacing `string` with typedefs?

**Comment by bboreham** [comment]
Sadly I had failed to observe it crashing very occasionally; there entries which have a hash key and range key but no value.  I have modified the code to be paranoid about all three.

**Comment by jml** [comment]
Still LGTM.

File: pkg/chunk/aws/storage_client.go [hunk 2]
```diff
@@ -795,6 +800,31 @@ func (b dynamoDBWriteBatch) TakeReqs(from dynamoDBWriteBatch, max int) {
...
+					hash = *hashAttr.S
+				}
+			}
+			var rnge, value []byte
+			if rangeAttr, ok := item[rangeKey]; ok {
+				rnge = rangeAttr.B
+			}
+			if valueAttr, ok := item[valueKey]; ok {
+				value = valueAttr.B
+			}
```
Comment (jml) [inline pkg/chunk/aws/storage_client.go]: Feels like something that could have been statically checked for. Oh well.

**Pull request merged**